### DISCUSSION TITLE
Get all topics with >0 subscribers matching a pattern

### DISCUSSION THREAD

**Discussion by gmencz** [uNetworking/uWebSockets.js#555] [Q&A] [answered]
The title says it all but I'll give an example use case. Each user in my app subscribes to their own topic (`user-123`, `user-456`, etc), is there a way to retrieve all topics which match a pattern? This would be useful to retrieve all topics which match the pattern `user-*` so that I can get all the online users.

**Comment by gitcatrat** [comment, marked as answer]
@gmencz I can't answer your question but...

How many "users" (I assume user === socket) do you have? Topics are relatively expensive, 1 topic per socket is not recommended, a much cheaper alternative is to store them in JS object/set like this:
```js
const sockets = {
  // some kind of user identifier as a key 
  'mS3lfSdF': SocketRef
};
```

And when you need to send a message to single socket, it's as easy as this:
```js
sockets[userId].send(message, isBinary, true);
```

**Reply by gmencz** [reply]
Yeah, from previous discussions I realized that 1 topic per socket is not ideal, however what led me to initially think that it was a good idea was this socket.io and uwebsockets.js comparison: https://alexhultman.medium.com/moving-from-socket-io-to-%C2%B5websockets-js-a85753b93a5a. It's talking about topics as if they were the same thing as rooms in socket.io, I get that conceptually they're the same thing but in socket.io, rooms are lightweight (a key in a Map) compared to a topic in uwebsockets so they shouldn't be directly compared like that since they can't be used in the same way.

Thank you for your suggestion, I ended up going with a Map.
